kqueen_ui/blueprints/ui/views.py

```diff
@@ -724,6 +732,42 @@ def handle(self, cluster_id):
         return redirect(request.environ.get('HTTP_REFERER', url_for('ui.index')))
 
 
+class ClusterSetPolicy(KQueenView):
+    decorators = [login_required]
+    methods = ['POST']
+    validation_hint = 'uuid'
+
+    def handle(self, cluster_id):
+        redirect_url = request.environ.get('HTTP_REFERER', url_for('ui.index'))
+
+        cluster = self.kqueen_request('cluster', 'get', fnargs=(cluster_id,))
+        policy = cluster['metadata'].get('network_policy', 'none')
+        if 'node_count' not in cluster.get('metadata', {}):
+            engine = cluster.get('provisioner', {}).get('engine', '<unknown>')
+            flash('{} engine doesn\'t support network policy.'.format(
+                prettify_engine_name(engine)), 'warning')
+            return redirect(redirect_url)
+
+        if policy.get('provider') == 'CALICO':
+            if policy.get('enabled'):
+                self.kqueen_request('cluster', 'set_network_policy', fnargs=(cluster_id, False))
+                flash('Network policy was successfully disabled', 'success')
+                return redirect(redirect_url)
+
+            if int(cluster['metadata']['node_count']) < 2:
+                flash('At least 2 nodes are required to enable network policy', 'error')
+                return redirect(redirect_url)
+
+            cluster = self.kqueen_request('cluster', 'set_network_policy',
+                                          fnargs=(cluster_id, True))
+            msg = 'Calico policy for cluster {} successfully enabled.'.format(cluster['name'])
+            user_logger.debug('{}:{}'.format(user_prefix(session), msg))
+            flash(msg, 'success')
+        else:
+            flash('Can not manage network policy for this cluster.', 'warning')
```

Comment:
It would be more descriptive to change the message like this:
`flash('Cannot manage network policy for {} network provider'.format(policy.get('provider', '<unknown>').capitalize())`

Reply:
This message appears when a user haven't set a policy during cluster creation. So provider is not known as no one was enabled